### PR TITLE
feat(ui): move select and deselect all actions to more intuitive checkbox

### PR DESCRIPTION
**Pull request by lavafroth**
### Changes
- Removes "Select all" and "Unselect All" button from the bottom row
- Adds a checkbox to the top row in line with the package checkbox entries for a more intuitive "select all" and "deselect all" experience

### CODE REVIEW

**Comment by Frigyes06** [comment]
Woag another UI contributor

**Comment by adhirajsinghchauhan** [comment]
While I like the idea, my main issue with it is that it's not perfectly aligned with other checkbox. It would need some manual tweaking.

Another minor issue is accessibility-related: the separate buttons at least conveyed their actions more obviously. Not everyone is familiar with the concept of a global checkbox. Adding an overlay on hover that conveys its meaning — based on state — would be better.

**Comment by lavafroth** [comment]
@adhirajsinghchauhan sure, I'll implement that.

**Comment by lavafroth** [comment]
Alright, should be good now. I also updated the screenshots.

File: src/gui/views/list.rs [hunk 1]
```diff
@@ -313,6 +315,12 @@ impl List {
                     .on_input(Message::SearchInputChanged)
                     .padding(5);
 
+                let select_all_checkbox =
+                    checkbox("Select all", self.all_selected, Message::ToggleAllSelected)
```
Comment (adhirajsinghchauhan) [inline src/gui/views/list.rs:319]: Tooltip would be better than a label, as it won't take up valuable space in the row. Also, text displayed should change based on `all_selected`.

I'll implement these changes myself.

**Review by adhirajsinghchauhan — CHANGES_REQUESTED** [review]

File: src/gui/views/list.rs (hunk 1 above)
Comment (adhirajsinghchauhan) [reply src/gui/views/list.rs]: Done (shown only on hover):

image

image

**Review by adhirajsinghchauhan — APPROVED** [review]

File: src/gui/views/list.rs (hunk 1 above)
Comment (lavafroth) [reply src/gui/views/list.rs]: Thanks 😃

**Pull request merged**